pkg/system/phase2_creating.go

```diff
@@ -723,15 +725,21 @@ func (r *Reconciler) ReconcileRootSecret() error {
 
 	// set noobaa root master key secret
 	if len(connectionDetails) != 0 {
+		// TODO: validate additional con details relevant to SA
 		if err := util.ValidateConnectionDetails(connectionDetails, authTokenSecretName, options.Namespace); err != nil {
 			return fmt.Errorf("ReconcileRootSecret: could not get/put key in external KMS: external kms connection details validation failed: %q", err)
 		}
 		kmsProvider := connectionDetails["KMS_PROVIDER"]
 		if util.IsVaultKMS(kmsProvider) {
-			keySecretName := "rootkeyb64-" + string(r.NooBaa.ObjectMeta.UID)
-			// reconcile root master key externally (vault)
-			c, err := util.InitVaultClient(connectionDetails, authTokenSecretName, options.Namespace)
-			if err == nil {
+			authMethod := connectionDetails["AUTH_METHOD"]
+			if(authMethod == "token") {
+				keySecretName := "rootkeyb64-" + string(r.NooBaa.ObjectMeta.UID)
```

Comment:
in the old token auth method the key is stored under NOOBAA_ROOT_SECRET_PATH, in here you removed it. I think we should keep it in the same place in this auth mehod too.